src/app/sensor-gui/sensor-gui.component.ts

```diff
@@ -0,0 +1,85 @@
+import { SensorData } from './../sensor-data';
+import { AfterViewInit, Component, Input, OnInit, ViewChild } from '@angular/core';
+import { Observable, Observer } from 'rxjs';
+import { BandThreeSensorService } from '../band-three-sensor.service';
+
+@Component({
+  selector: 'app-sensor-gui',
+  templateUrl: './sensor-gui.component.html',
+  styleUrls: ['./sensor-gui.component.css']
+})
+export class SensorGuiComponent implements OnInit, AfterViewInit, Observer<BandThreeSensorService> {
+
+  @ViewChild("knob") knob;
+
+  observable: Observable<SensorData>;
+  sensorData: SensorData;
+  dataArray: SensorData[] = [];
+
+  constructor(private band3: BandThreeSensorService) {
+    this.sensorData = new SensorData();
+    this.sensorData.timestamp = new Date();
+    this.sensorData.temperature = 0.0;
+  }
+
+  click() {
+    this.knob.checked = !(this.knob.checked);
+    console.log("knob: " + this.knob.checked);
+  }
+
+  ngOnInit(): void {
+    console.log("oninit");
+  }
+
+  ngAfterViewInit(): void {
+    this.observable = this.band3.sensor;
+    this.observable.subscribe(this);
+    this.knob.checked = true;
+  }
+
+  next(data) {
+    if (this.knob.checked) {
+      this.dataArray.push(data);
+      this.sensorData.timestamp = data.timestamp.toISOString();
+      this.sensorData.temperature = data.temperature;
+      this.sensorData.averageTemp = this.calculateAverage(5000);//millseconds
+    }
+  }
+
+  calculateAverage(interval: number): number {
```

Comment:
I don't know whether this computes the correct average, I assume so. 
As I suggested in the problem description hints ("You may want to use ... a combination of s`ampleTime()`, `map()` and `scan()` to compute the running average"): instead of devising your own algorithm you could have employed the `scan()` operator to extract "slices" of 5 consecutive values from the stream, then `map()` to convert each slice to its average. You can take a look at `printableAverageStream()` in  _https://github.com/amchavan/angular-training/blob/unit1L-exercises/src/app/app.component.ts_

Let me please explain that I'm not insisting with using RxSJ operators because I believe they are always the better option. It may well be that in this specific case your solution is better (however you want to define "better"). However, one needs to know what operators are available and when you can use them, and this exercise was a means for that end. (I'm sure you can compute a moving average without extra training!)